### PR TITLE
Added settings page UI

### PR DESCRIPTION
**Pull request by sherry1iu**
image

Adds non-functional UI for settings, shown above. Opens after pressing settings button in sidebar, closes upon entering another tab

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,24 +14,32 @@
...
     "maplibre-gl": "^2.4.0",
+    "prop-types": "^15.8.1",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "react-map-gl": "^7.0.19",
     "react-redux": "^8.0.4",
-    "redux-saga": "^1.2.1"
+    "react-router-dom": "^6.4.3",
+    "redux-saga": "^1.2.1",
+    "tileserver-gl-light": "^4.1.2"
```
Comment (ajmcquilkin) [inline package.json:34]: This isn't relevant to this PR

File: package.json [hunk 2]
```diff
@@ -14,24 +14,32 @@
...
   },
   "dependencies": {
-    "@heroicons/react": "^2.0.12",
+    "@heroicons/react": "^2.0.13",
+    "@material-tailwind/react": "^1.2.4",
     "@meshtastic/meshtasticjs": "^0.6.113",
     "@reduxjs/toolkit": "^1.8.6",
     "@tauri-apps/api": "^1.1.0",
+    "@types/react-router": "^5.1.19",
+    "all": "^0.0.0",
```
Comment (ajmcquilkin) [inline package.json:23]: What is this package?

File: src/App.tsx [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import React from "react";
-import Sidebar from "@components/Sidebar/Sidebar";
-import { MapView } from "@components/Map/MapView";
+import Sidebar from "./Components/Sidebar/Sidebar";
+import { MapView } from "./Components/Map/MapView";
+import { Route, Routes } from "react-router-dom";
```
Comment (ajmcquilkin) [inline src/App.tsx:4]: Nit: would place this below the `react` import

File: src/components/Map/MapView.tsx [hunk 4]
```diff
@@ -26,6 +27,12 @@ const _MapViewLeftPanel = ({ activeSidebarPanel }: _IMapViewLeftPanel) => {
     case "map":
       return <NodeSearchDock />;
 
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (ajmcquilkin) [reply src/components/Map/MapView.tsx]: Second this

File: src/components/NodeSearch/NodeSearchDock.tsx [hunk 5]
```diff
@@ -64,8 +80,12 @@ const NodeSearchDock = () => {
       const lowercaseQuery = query.toLocaleLowerCase();
 
       if (String(node.data.num).includes(lowercaseQuery)) return true;
-      if (node.data.user?.shortName.toLocaleLowerCase().includes(lowercaseQuery)) return true;
-      if (node.data.user?.longName.toLocaleLowerCase().includes(lowercaseQuery)) return true;
+      if (
```
[the remark this answers is not in the document]
Comment (sherry1iu) [reply src/components/NodeSearch/NodeSearchDock.tsx:83]: I just prettiered this one, I'll make an issue for it

**Review by ajmcquilkin — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 6]
```diff
@@ -24,3 +24,4 @@ dist-ssr
 *.sw?
 node_modules
 
+*.mbtiles
```
Comment (ajmcquilkin) [inline .gitignore:27]: We shouldn't have this in a non-map PR

File: src/components/Map/MapView.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (ajmcquilkin) [reply src/components/Map/MapView.tsx]: Outdated

**Review by ajmcquilkin — APPROVED** [review]
Looks good, nice work!

**Pull request merged**